lib/internal/Magento/Framework/Controller/Test/Unit/Result/RedirectTest.php

```diff
@@ -67,16 +70,44 @@ public function testSetPath()
             $this->returnValue($params)
         );
         $this->assertInstanceOf(
-            \Magento\Framework\Controller\Result\Redirect::class,
+            Redirect::class,
             $this->redirect->setPath($path, $params)
         );
     }
 
-    public function testRender()
+    public function testRenderDataProvider()
     {
-        $this->response->expects($this->once())->method('setRedirect');
+        return [
+            ['http://test.com', null, 302],
+            ['http://test.com', 0, 302],
+            ['http://test.com', '', 302],
+            ['http://test.com', 301, 301],
+            ['http://test.com', 302, 302],
+            ['http://test.com', 303, 303],
+            ['http://test.com', 200, 302],
+            ['http://test.com', 404, 302],
+            ['http://test.com', 503, 302],
+        ];
+    }
+
+    /**
+     * @param string $url
+     * @param mixed $statusCode
+     * @param int $expectedStatusCode
+     * @dataProvider testRenderDataProvider
+     */
+    public function testRender($url, $statusCode, $expectedStatusCode)
+    {
+        $this->redirect->setUrl($url);
+        $this->redirect->setHttpResponseCode($statusCode);
+
+        $this->response
+            ->expects($this->once())
+            ->method('setRedirect')
+            ->with($this->equalTo($url), $this->equalTo($expectedStatusCode));
+
         $this->assertInstanceOf(
-            \Magento\Framework\Controller\Result\Redirect::class,
+            Redirect::class,
             $this->redirect->renderResult($this->response)
```

Comment:
Please remove this pretty useless assert and simply call the method instead.

Reply:
I did remove it, but are you sure this assert is useless? removing the return from renderResult could possibly lead to breaking things if something outside depends on the return.

Reply:
Mainly because I wouldn't treat `return $this` as class behavior but rather a particular implementation details :) Just assume that for every `return $this` in code we must have an assertion in some test. I believe it would be better to control it with dedicated static test and it matters in unit test only in scope of some bugfix, when return was previously missing.

I do understand your opinion however and removing "useless" (as to me) assert obviously does not deserve a separate commit.

Reply:
gotcha, thanks for the explanation!